torch/distributed/_composable/replicate.py

```diff
@@ -11,59 +12,30 @@
 _ROOT_MODULE_PREFIX = ""
 
 
-@contract()
-def replicate(
-    module: nn.Module,  # NOTE: contract now supports single module only
-    ignored_modules: Optional[Iterable[torch.nn.Module]] = None,
-    **kwargs,
-) -> nn.Module:
-    r"""Replicates a module
-
-    Args:
-        module (torch.nn.Module): module to replicate
-
-    Example::
-        >>> # xdoctest: +REQUIRES(module:torch._C._distributed_c10d)
-        >>> module = nn.Linear(3, 3)
-        >>> replicate(module)
-    """
-    torch._C._log_api_usage_once("torch.distributed.replicate")
-    if "device_id" in kwargs:
-        if not isinstance(kwargs["device_id"], (int, torch.device)):
-            raise RuntimeError(
-                f"Expected device_id to be int or torch.device, but got {type(kwargs['device_id'])}"
-            )
-    _ReplicateState(ignored_modules=ignored_modules).mark_module(module, **kwargs)
-    return module
-
-
-def _is_fully_sharded(module: nn.Module) -> bool:
-    r"""Check if module is marked with fully_shard."""
-    return "fully_shard" in _get_registry(module)
-
-
-class _ReplicateState:
-    def __init__(self, ignored_modules: Optional[Iterable[torch.nn.Module]]) -> None:
+class _ReplicateState(_State):
+    def __init__(self) -> None:
```

Comment:
no call to super()? 

what's the reason to subclass _State

Reply:
I believe the reason is that we are using the _State for composable APIs. Correct me if I am wrong.